### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by HarshitJoshi**
Updated tests based on review and added more tests

### CODE REVIEW

File: packages/bitcore-node/test/unit/models/block.unit.ts [hunk 1]
```diff
@@ -104,72 +95,215 @@ describe('Block Model', function() {
...
-      sandbox.stub(CoinModel, 'update').resolves();
-      sandbox.stub(BlockModel, 'findOne').returns({
-        sort: sandbox.stub().resolves(null)
+
+    it('should return if localTip hash equals the previous hash', async () => {
+      let blockModelRemoveSpy = sandbox.stub(BlockModel, 'remove').resolves();
+      let transactionModelRemoveSpy = sandbox.stub(TransactionModel, 'remove').resolves();
+      let coinModelRemoveSpy = sandbox.stub(CoinModel, 'remove').resolves();
+      let coinModelUpdateSpy = sandbox.stub(CoinModel, 'update').resolves();
+      let blockModelFindOneSpy = sandbox.stub(BlockModel, 'findOne').returns({
```
[the remark this answers is not in the document]
Comment (HarshitJoshi) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: resolved

File: packages/bitcore-node/test/unit/models/block.unit.ts [hunk 2]
```diff
@@ -104,72 +95,215 @@ describe('Block Model', function() {
...
+      expect(transactionModelRemoveSpy.calledOnce).to.be.true;
+
+    });
+
+    it('should call coinModel remove', async () => {
+      sandbox.stub(BlockModel, 'remove').resolves();
+      sandbox.stub(TransactionModel, 'remove').resolves();
+      let coinModelRemoveSpy = sandbox.stub(CoinModel, 'remove').resolves();
+      sandbox.stub(CoinModel, 'update').resolves();
+      sandbox.stub(BlockModel, 'findOne').returns({
```
[the remark this answers is not in the document]
Comment (HarshitJoshi) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: resolved

File: packages/bitcore-node/test/unit/models/block.unit.ts [hunk 3]
```diff
@@ -104,72 +95,215 @@ describe('Block Model', function() {
...
+      expect(blockModelRemoveSpy.calledOnce).to.be.true;
+
+    });
+
+    it('should call transactionModel remove', async () => {
+      sandbox.stub(BlockModel, 'remove').resolves();
+      let transactionModelRemoveSpy = sandbox.stub(TransactionModel, 'remove').resolves();
+      sandbox.stub(CoinModel, 'remove').resolves();
+      sandbox.stub(CoinModel, 'update').resolves();
+      sandbox.stub(BlockModel, 'findOne').returns({
```
[the remark this answers is not in the document]
Comment (HarshitJoshi) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: resolved

File: packages/bitcore-node/test/unit/models/block.unit.ts [hunk 4]
```diff
@@ -104,72 +95,215 @@ describe('Block Model', function() {
...
+      expect(blockModelFindOneSpy.notCalled).to.be.false;
+
+    });
+
+    it('should call blockModel remove', async () => {
+      let blockModelRemoveSpy = sandbox.stub(BlockModel, 'remove').resolves();
+      sandbox.stub(TransactionModel, 'remove').resolves();
+      sandbox.stub(CoinModel, 'remove').resolves();
+      sandbox.stub(CoinModel, 'update').resolves();
+      sandbox.stub(BlockModel, 'findOne').returns({
```
[the remark this answers is not in the document]
Comment (HarshitJoshi) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: resolved

File: packages/bitcore-node/test/unit/models/block.unit.ts [hunk 5]
```diff
@@ -104,72 +95,215 @@ describe('Block Model', function() {
...
+      expect(blockModelFindOneSpy.notCalled).to.be.false;
+
+    });
+
+    it('should return if localTip height is zero', async () => {
+      let blockModelRemoveSpy = sandbox.stub(BlockModel, 'remove').resolves();
+      let transactionModelRemoveSpy = sandbox.stub(TransactionModel, 'remove').resolves();
+      let coinModelRemoveSpy = sandbox.stub(CoinModel, 'remove').resolves();
+      let coinModelUpdateSpy = sandbox.stub(CoinModel, 'update').resolves();
+      let blockModelFindOneSpy = sandbox.stub(BlockModel, 'findOne').returns({
```
[the remark this answers is not in the document]
Comment (HarshitJoshi) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: resolved

File: packages/bitcore-node/test/unit/models/block.unit.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanielRHayes) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: Are you missing a commit? It still looks like `BlockModel.findOne` is stubbed?

File: packages/bitcore-node/test/unit/models/block.unit.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (DanielRHayes) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: Are you missing a commit? It still looks like `BlockModel.findOne` is stubbed.

File: packages/bitcore-node/test/unit/models/block.unit.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (DanielRHayes) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: Are you missing a commit? It still looks like `BlockModel.findOne` is stubbed.

File: packages/bitcore-node/test/unit/models/block.unit.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (DanielRHayes) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: Are you missing a commit? It still looks like `BlockModel.findOne` is stubbed.

File: packages/bitcore-node/test/unit/models/block.unit.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (DanielRHayes) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: Are you missing a commit? It still looks like `BlockModel.findOne` is stubbed.

File: packages/bitcore-node/test/unit/models/block.unit.ts [hunk 6]
```diff
@@ -104,72 +95,215 @@ describe('Block Model', function() {
...
+      expect(coinModelRemoveSpy.calledOnce).to.be.true;
+
+    });
+
+    it('should call coinModel update', async () => {
+      sandbox.stub(BlockModel, 'remove').resolves();
+      sandbox.stub(TransactionModel, 'remove').resolves();
+      sandbox.stub(CoinModel, 'remove').resolves();
+      let coinModelUpdateSpy = sandbox.stub(CoinModel, 'update').resolves();
+      sandbox.stub(BlockModel, 'findOne').returns({
```
[the remark this answers is not in the document]
Comment (DanielRHayes) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: Are you missing a commit? It still looks like `BlockModel.findOne` is stubbed.

File: packages/bitcore-node/test/unit/models/block.unit.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (HarshitJoshi) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: should be fixed now

File: packages/bitcore-node/test/unit/models/block.unit.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (HarshitJoshi) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: should be fixed now

File: packages/bitcore-node/test/unit/models/block.unit.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (HarshitJoshi) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: should be fixed now

File: packages/bitcore-node/test/unit/models/block.unit.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (HarshitJoshi) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: should be fixed now

File: packages/bitcore-node/test/unit/models/block.unit.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (HarshitJoshi) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: should be fixed now

[the remark this answers is not in the document]
Comment (HarshitJoshi) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: should be fixed now

File: packages/bitcore-node/test/unit/models/block.unit.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (HarshitJoshi) [reply packages/bitcore-node/test/unit/models/block.unit.ts]: should be fixed now

**Comment by DanielRHayes** [comment]
Looks good, we can merge it

File: packages/bitcore-node/test/unit/models/block.unit.ts [hunk 7]
```diff
@@ -68,11 +64,7 @@ describe('Block Model', function() {
   });
 
   describe('getPoolInfo', () => {
-    it('UNIMPLEMENTED: should return pool info given a coinbase string', () => {
-      expect(() => {
-        BlockModel.getPoolInfo('');
-      }).to.not.throw(TypeError);
-    });
+    xit('should return pool info given a coinbase string');
```
Comment (micahriggan) [inline packages/bitcore-node/test/unit/models/block.unit.ts]: xit?

**Pull request merged**